src/streamers/spotify.streamer.ts

```diff
@@ -1,146 +1,200 @@
 
     if (!jsonData.is_playing || jsonData.currently_playing_type !== 'track') {
-      return this.fetchLastPlayed(accessToken);
+      return this.fetchLastPlayed(accessToken)
     }
 
-    const track = jsonData.item;
+    const track = jsonData.item
 
     const song: Song = {
       is_playing: jsonData.is_playing,
       title: track.name,
-      artiste: track.artists.map(artist => artist.name).join(', '),
+      artiste: track.artists.map((artist) => artist.name).join(', '),
       image_url: track.album.images[0].url,
       preview_url: track.preview_url,
       url: track.external_urls.spotify,
-    };
+    }
 
     if (this.useCache) {
-      this.storer.set(SPOTIFY_TRACK_KEY, song, this.cacheDuration);
+      this.storer.set(SPOTIFY_TRACK_KEY, song, this.cacheDuration)
     }
-    return song;
+    return song
   }
 
-  private async fetchLastPlayed(accessToken: SpotifyAccessToken): Promise<Song | null> {
-    const headers = { Authorization: `Bearer ${accessToken.access_token}`, 'Content-Type': 'application/json', Accept: 'application/json' };
-    const response: Response = await fetch('https://api.spotify.com/v1/me/player/recently-played?limit=1', { method: 'GET', headers });
-    const jsonData = await response.json() as SpotifyRecentlyPlayedResponse;
+  async fetchLastPlayed(accessToken: SpotifyAccessToken): Promise<Song | null> {
+    const headers = {
+      Authorization: `Bearer ${accessToken.access_token}`,
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    }
+    const response: Response = await fetch(
+      'https://api.spotify.com/v1/me/player/recently-played?limit=1',
+      { method: 'GET', headers }
+    )
+    if (!response.ok) {
+      const errorInfo = (await response.json()) as SpotifyErrorResponse
+      return Promise.reject(errorInfo.error)
+    }
+    const jsonData = (await response.json()) as SpotifyRecentlyPlayedResponse
 
-    if (jsonData.items.length === 0) {
-      return null;
+    if (jsonData.items.length === 0 || jsonData.total === 0) {
+      return null
     }
 
-    const [{ track: lastPlayedTrack }] = jsonData.items;
+    const [{ track: lastPlayedTrack }] = jsonData.items
     const lastPlayed = {
       is_playing: false,
       title: lastPlayedTrack.name,
-      artiste: lastPlayedTrack.artists.map(artist => artist.name).join(', '),
+      artiste: lastPlayedTrack.artists.map((artist) => artist.name).join(', '),
       image_url: lastPlayedTrack.album.images[0].url,
       preview_url: lastPlayedTrack.preview_url,
       url: lastPlayedTrack.external_urls.spotify,
-    };
+    }
 
     if (this.useCache) {
-      this.storer.set(SPOTIFY_TRACK_KEY, lastPlayed, this.cacheDuration);
+      this.storer.set(SPOTIFY_TRACK_KEY, lastPlayed, this.cacheDuration)
     }
 
-    return lastPlayed;
+    return lastPlayed
+  }
+
+  setUseCache(useCache: boolean): void {
+    this.useCache = useCache
```

Comment:
This function is not in use.

Reply:
I use it in the test to change the value of `useCache`. Check https://github.com/BolajiOlajide/now-playing/blob/bo/add-tests-configure-prettier/src/streamers/__tests__/spotify.streamer.spec.ts#L276